src/highlighters.cc

```diff
@@ -1174,6 +1187,7 @@ struct LineNumbersHighlighter : Highlighter
     const bool m_relative;
     const bool m_hl_cursor_line;
     const String m_separator;
+    const Optional<String> m_separator_cursor;
```

Comment:
Thinking a bit more about it, I think we could avoid the Optional here and just duplicate the separator is -separator-cursor was not provided. Separators are very likely to fit into the small string optimization and it would simplify the highlighting logic by not having to care about whether that optional is engaged or not.

Reply:
That's how I actually implemented everything initially. I thought that it's a bit weird to have two strings that will have the same value for most use cases, but the implementation is indeed a lot simpler. Should we revert back to using just `String`?

Reply:
Yeah, I think we can go with the simplest implementation, especially as the Optional does not buy us any perf/memory.